test/creditguild-test.js

```diff
@@ -0,0 +1,47 @@
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+
+describe("CreditGuild", function () {
+    
+    let membershipFee = 20;
+    let vouchAmout = 1;
+    let marketRegistry = "0x4ABAb00eae341cF2E79D2ae35455749F10B39dEC";
```

Comment:
Deploy this mock contract:

```solidity
pragma solidity 0.8.4;

contract MarketRegistryMock {
    struct Market {
        address uToken;
        address userManager;
    }

    address[] public uTokenList;
    address[] public userManagerList;
    mapping(address => Market) public tokens;

    function addUToken(address token, address uToken) public {
        uTokenList.push(uToken);
        tokens[token].uToken = uToken;
    }

    function addUserManager(address token, address userManager) public {
        userManagerList.push(userManager);
        tokens[token].userManager = userManager;
    }
}
```

Reply:
After you deploy this you also need to deploy a mock UserManager and mock UToken for uDAI.
Then you need to call:

```solidity
// this adds the mockUtoken contract for the mock underlying token (DAI)
await marketRegistryMock.addUToken(underlying.address, mockUToken.address);
```